### DISCUSSION TITLE
power on coffee maker

### DISCUSSION THREAD

**Discussion by oxedgar** [alexkn/node-red-contrib-homeconnect#23] [Q&A] [answered]
Hello, Sorry but i cannot hable to make your node working!
i can connect to my appliance fine because i can get HSID form my node-red server! Is that my connection is working! but only get_appliances is working!

Wehn i try to make onther thing i together get this error:
`{"payload":"xxxxf2c9.6d27","event":{"clientX":579,"clientY":193,"bbox":[499,227,655,179]},"socketid":"EgSjKkxxxxxxy4cCAAAE","_msgid":"8bxxx9b9.29b978","error":"Error: Conflict"}`
Have you any idee?

Can you send me any screenshot of working config has exemple?
Many thanks for your job!

**Comment by alexkn** [comment]
What are you trying to do?
Have you authorized the correct scopes?

**Comment by oxedgar** [comment]
Hello,



I just ant to power on my coffee machine at specific times of the day! 

For try, I have added on the UI one trigger button to start action and just send the result on debug console: 

 

I get badrequest or conflict message from HomeConnect. Only get_Home_Appliances is working!


I think my scopes is correct and I cannot find any way to change it:

**Reply by alexkn** [reply]
I moved the issue to discussions.
The issue tracker is for bug reports, not for questions.

Please describe exactly what you tried. Maybe then I can help you.

Please answer on github, email answers are messy.

**Comment by oxedgar** [comment]
I followed the account creation operations as you indicated in the readme file!

In node red I created several buttons to simulate my commands but none of them seem to work correctly! the message back in the debug node is either conflict or bad request!

Communication is working because I can use get_Home_Appliances and get_Specific_Appliance functions.

I only have one device (Siemens coffee machine)!
When I try to get the status of it, I get a conflict message!

Do you have any idea what can block the working of these functions (which do not require specific parameters from me)?

Also, I would be grateful if you could provide me with configuration screenshots as examples for configuring commands like pose on, standby, run a program or any description! Do you have a how to guide?

This is a wakeup fonction with result of call it in debug is not working:
![Problème POwerOn](https://user-images.githubusercontent.com/19332204/105241540-1ca46c80-5b6e-11eb-9d4d-dbab3b341b5a.png)

This is getState function result is not working:
![GetStates](https://user-images.githubusercontent.com/19332204/105242947-a6ecd080-5b6e-11eb-9410-191e957e4112.png)

This is the result of getHomeAppliances is working:
![GetHomeAppliance](https://user-images.githubusercontent.com/19332204/105243156-f16e4d00-5b6e-11eb-87e3-fa2f6ca829a0.png)

**Comment by alexkn** [comment, marked as answer]
set_setting:
The settingskey should be "BSH.Common.Setting.PowerState" without the ".Standby" at the end.
Maybe you need more in the body, look at the api example:
https://apiclient.home-connect.com/#/settings/set_setting

get_status:
I don't see whats wrong.
Maybe you don't have authorized the needed scope.
What scoped do you have authorized?

**Reply by oxedgar** [reply]
I cannot find where set the scopes! iWhen i created my appliance, it is just asked for:
-application name, i've set "one name"
-OAuth Flow, i've selected "Authorization code grant Flow"
-Home Connect User Account for Testing, i've set my homeConnect account mail
-Redirect URI, I've set the url of my node-red server
-One Time Token, is not checked

After that, my application is created and i get my app ID and my SecretID!
In edit of application, i can view one table with scopes where all is checked and i cannot edit rights! It is your the same?
Have you direct url to edit scopes or others?

And for the body, i can see this on the homeConnect api:
`{
  "data": {
    "key": "BSH.Common.Setting.PowerState",
    "value": "BSH.Common.EnumType.PowerState.On",
    "type": "BSH.Common.EnumType.PowerState",
    "constraints": {
      "allowedvalues": [
        "BSH.Common.EnumType.PowerState.On",
        "BSH.Common.EnumType.PowerState.Standby"
      ]
    }
  }
}`

I need to put all this exactly like this or i need to change it? if i need to change, wath i need to put where?

**Reply by alexkn** [reply]
If you see the scopes all checked in the edit application view, it should be already right.

You can set the scopes in the auth node before starting authorization.
If you leave it empty it should request all scopes.
You need at least: 
IdentifyAppliance CoffeeMaker

**Reply by oxedgar** [reply]
i have this scopes IdentifyAppliance CoffeeMaker and only this is listed wehn i make autorisation!

**Reply by oxedgar** [reply]
i found issue with connection of coffee machine! It is not connected using wifi to router but only with wifi with my phone!!! i delete machine from my HomeConnect account and reset network configuration on machine. After reconnect again, the get_Status work fine and standby and powerOn is working now. 

Thank you for helping
